### ISSUE TITLE
IMqttEntityManager logging is too verbose in debug mode

### ISSUE THREAD

**Issue by rumbu13** [net-daemon/netdaemon#723]
## The problem

It is nearly impossible to debug a NetDaemon App due to verbosity of messages logged in console/log file. Every MQTT message generated by your app is logged, making extremely difficult to discern between your own debug messages and the ones generated by ```IMqttEntityManager```.

## Use case

I am updating 4 different sensors with rssi values gathered from various beacons. Since the beacons send their values usually every 3 seconds, there is a call to ```SetStateAsync/SetAttributeAsync``` every 3/4 seconds. That means that during 1 minute the screen/log is filled with 80-160 messages.

MQQT has a high throughoutput by design, it is expected to have a high number of messages but there is no need to have them as debug information.  

A better option would be to log these messages with ```TRACE``` level instead of ```DEBUG``` level. Irrespective of the log level, logging every MQTT message can lead to a performance bottleneck. People really interested in ALL messages sent by their app can use a dedicated tool like MQTT Explorer or make their own calls to ```LogDebug``` when required.

## Environment

- NetDaemon release with the issue: 22.23.0
- Operating environment (Home assistant Add-on/Docker/Dev setup): 2022.6.7

**Comment by skotl** [comment]
Fair point. The only excuse I have is that it's still in R&D mode however I think your point is valid.
I'm happy to create a PR for this, and equally happy if you want to do so.

As a workaround, ND currently uses a fairly simple logging configuration but you can very easily replace it with your own and  decrease the level of logging for `NetDaemon.Extensions.MqttEntityManager`. I just tested this and the whole exercise took me 3 minutes, plus you get the benefit of full SeriLog configuration.

The docs to do so are over here: https://netdaemon.xyz/docs/v2/app_model/app_model_custom_logging#step-1---create-a-new-logging-configuration-class

And your appsettings.json will end up with a line that reads

```json
 "Serilog": {
    "MinimumLevel": {
      "Default": "Debug",
      "Override": {
        "System": "Information",
        "Microsoft": "Information",
        "System.Net.Http.HttpClient": "Warning",
        "daemonapp.apps.ScottHome": "Verbose",
        "NetDaemon.Extensions.MqttEntityManager": "Information"
        ...```

**Comment by skotl** [comment]
OP confirmed that the workaround resolved his immediate issue, but I think the request is valid (especially given our intent to ship a usable platform out of the box).
I'll turn the majority of the logging down to trace as part of the mqttnet v4 PR.

**Comment by helto4real** [comment]
V4 is released. Can we close this now?

**Comment by skotl** [comment]
Oops - never got back to it.

I'll log a PR shortly with the tests for Rumbu's change and also dialling down the logging, then I'll close this issue.

I don't think this is critical and can wait until the next release, as the OP has a workaround.

**Comment by skotl** [comment]
Closing as will be addressed when #725 is merged
